Jenkinsfile

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env groovy
+
+pipeline {
+    agent { docker defaultWorker.getConfig("openstack-test") }
+    options {
+        ansiColor('xterm')
+        timestamps()
+        timeout(time: 2, unit: "HOURS")
+    }
+    stages {
+        stage("systest") {
+            steps {
+                sh '''
+                    # - initialize job control vars
+                    # - JOB_NAME is an environment variable that is
+                    #   automatically provided by Jenkins
+                    #   (eg. "openstack/driver/mitaka/12.1.1-overcloud-vxlan")
+                    export PROGRAM=$(echo $JOB_NAME | cut -d "/" -f 1)
+                    export PROJECT=$(echo $JOB_NAME | cut -d "/" -f 2)
+                    export BRANCH=$(echo $JOB_NAME | cut -d "/" -f 3)
```

Comment:
Jenkins job names cannot contain forward slashes (Jenkins won't let you create them).  So this cut should work as expected.

When you add a Jenkins folder to track the jobs for "stable/newton" you'll have to pick Jenkins-frienly name (like "stable-newton" or "stable_newton").